### PR TITLE
DO NOT MERGE: Fix Terraform help printing log twice and causing error

### LINKED ISSUE #781: validate command prints usage when used like documented?
### Describe the Bug

We bumped from 1.90.0 to 1.105.0. Our CI uses 

`atmos validate component <component> --stack <stack>`

With 1.105.0 (and 1.106.0) we now get the commands usage printed instead/additionally:

```
atmos validate component "vpc/foo" --stack bar
```

```
Atmos supports native '' commands with all the options, arguments and flags.

In addition, 'component' and 'stack' are required in order to generate variables for the component in the stack.

atmos  <subcommand> <component> -s <stack> [options]
atmos  <subcommand> <component> --stack <stack> [options]

For more details, execute ' --help'

component 'vpc/foo' in stack 'bar' validated successfully
```

So it looks like some validation still took place and it was successfull.

### Expected Behavior

Exit code 0 on successfull validation, some other number otherwise. No usage gets printed to the screen.

### Steps to Reproduce

Run `atmos validate component <component> --stack <stack>` as documented via `--help`.

### Screenshots

_No response_

### Environment

- OS: Linux
- atmos Version: 1.105.0
- terraform Version: 1.8.5

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Cerebrovinny**
## what

This fix Terraform help command causing logo to display twice and causes error

## Evidences Test
Before:
Screenshot 2024-11-21 at 15 15 09

After 2 flags testing (--help) and (help):
Screenshot 2024-11-21 at 23 31 30

Screenshot 2024-11-21 at 23 31 48

## references
- Closes #781 

## Summary by CodeRabbit

- **New Features**
	- Enhanced command execution flow for help commands, allowing for early exits and improved user experience when invoking help.
	- Streamlined command-line argument handling, enabling recognition of help commands without requiring additional parameters.
	- Introduced a new help flag (`-help`) for improved command-line assistance.
	- Simplified logo printing process with a dedicated function for the Atmos logo.

- **Bug Fixes**
	- Improved error handling and validation for command execution, ensuring necessary fields are checked before processing commands.
	- Enhanced control flow and error handling in the command execution process.

- **Documentation**
	- Clarified comments regarding command-line argument processing and component configurations.

### CODE REVIEW

**Comment by osterman** [comment]
@Cerebrovinny can you please post screenshots before/after.

**Comment by Cerebrovinny** [comment]
@osterman @aknysh this also fixes Atmos workflows showing terraform help

CleanShot 2024-11-22 at 22 15 14

File: internal/exec/utils.go [hunk 1]
```diff
@@ -285,6 +281,16 @@ func ProcessStacks(
 
 	configAndStacksInfo.StackFromArg = configAndStacksInfo.Stack
 
+	// Initialize component section maps
```
[the remark this answers is not in the document]
Comment (Cerebrovinny) [reply internal/exec/utils.go:284]: ProcessComponentConfig as causing it to display nil pointer dereference errors in tests when we process the stack

File: internal/exec/terraform.go [hunk 2]
```diff
@@ -56,16 +52,31 @@ func ExecuteTerraform(info schema.ConfigAndStacksInfo) error {
...
 	}
 
+	// For all other commands, we need to process stacks
+	cliConfig, err := cfg.InitCliConfig(info, true)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println()
+	err = tuiUtils.PrintStyledText("ATMOS")
```
[the remark this answers is not in the document]
Comment (Cerebrovinny) [reply internal/exec/terraform.go:66]: guess this is a question to @osterman

[the remark this answers is not in the document]
Comment (osterman) [reply internal/exec/terraform.go:66]: We should not print the ATMOS logo on any command other than help. 

https://github.com/cloudposse/atmos/pull/793/files#diff-7bed7f9ea19fe4d3abd5614c8a15877c0148d79a8b7cd1275e549e0f65589d17R88

[the remark this answers is not in the document]
Comment (Cerebrovinny) [reply internal/exec/terraform.go]: resolved this

**Comment by osterman** [comment]
Per the conventions here: 
- https://github.com/cloudposse/atmos/pull/793/files#diff-7bed7f9ea19fe4d3abd5614c8a15877c0148d79a8b7cd1275e549e0f65589d17R88

We should have the logo on all `--help` commands, but not on usage emitted on errors.

image


This output feels redundant.

This is repeated twice.
> unknown command "asd" for "atmos"
>  Did you mean this?
>	aws



image

**Comment by aknysh** [comment]
@Cerebrovinny  please resolve the conflicts and address the comments

**Comment by Cerebrovinny** [comment]
Screenshot 2024-12-05 at 17 18 39
Screenshot 2024-12-05 at 17 19 45
Screenshot 2024-12-05 at 17 20 08

**Comment by osterman** [comment]
@Cerebrovinny looks like the columns are still pinched.

**Comment by Cerebrovinny** [comment]
> We should have the logo on all `--help` commands, but not on usage emitted on errors.

This is another PR that does not contain those fixes. Perhaps you would like to merge everything into a single PR where we can also address the style issues?

**Comment by Cerebrovinny** [comment]
> @Cerebrovinny looks like the columns are still pinched.

@osterman This is another PR that does not contain those fixes. Perhaps you would like to merge everything into a single PR where we can also address the style issues?

**Comment by aknysh** [comment]
@Cerebrovinny 

when running `atmos terraform --help`, two logos are still printed

when running `atmos terraform help`, no logo is printed

can you please review it again

**Comment by Cerebrovinny** [comment]
> atmos terraform help

@aknysh thank you, still need to push changes for this case, I am also integrating tests to make sure is correct

**Comment by osterman** [comment]
@Cerebrovinny I was confused. You can disregard.


 @aknysh brings up a good point that logo no longer is displayed on help.

**Comment by osterman** [comment]
Let's close this for now. We can reopen if we still need it.

**Pull request closed without merging**